pkg/controller/deployment/deployment_controller.go

```diff
@@ -637,6 +734,15 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return dc.sync(d, rsList, podMap)
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.VerticalScaling) {
+		hasResourceUpdate, _ := dc.patchDeploymentResource(d)
+
+		// return early to prevent creating new replicas when doing resource update
+		if hasResourceUpdate {
```

Comment:
If there are any changes in addition to resources update, we should continue.

Reply:
this will be fixed in subsequent PR